### PR TITLE
`General`: Only show learning goals when data is available

### PR DESCRIPTION
**Pull request by ge65cer**
#### General

- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] I chose a title conforming to the [naming conventions for pull requests](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/development-process.html#naming-conventions-for-github-pull-requests).
#### Server
- [x] I followed the [coding and design guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/server/).
- [x] I implemented the changes with a good performance and prevented too many database calls.
#### Client
- [x] I followed the [coding and design guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client/) and ensured that the layout is responsive.
- [x] Following the [theming guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client-design/), I specified colors only in the theming variable files and checked that the changes look consistent in both the light and the dark theme.
- [x] I added multiple integration tests (Jest) related to the features (with a high test coverage), while following the [test guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client-tests/).
- [x] I documented the TypeScript code using JSDoc style.

### Motivation and Context

The learning goals tab is shown, even when the instructor had not defined any learning goals or prerequisites for the course.

### Description

This PR fixes this issue by only showing the learning goals tab when data is available.
The learning goals are only loaded in the course detail view, not in the whole overview, to improve performance.

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Course with Learning Goals/Prerequisites

1. Log in to Artemis
2. Navigate to Course Overview
3. Ensure that the learning goals tab is displayed
4. Navigate to Course Administration
5. Delete all learning goals/prerequisites of the course
6. Navigate back to Course Overview
7. Ensure the learning goals tab not shown

#### Code Review
- [x] Review 1
- [x] Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

### Test Coverage

| Class/File | Branch | Line |
|------------|-------:|-----:|
| course-overview.component.ts | 83% | 93% |

### CODE REVIEW

**Review by jbllhff — APPROVED** [review]
Tested on ts2, works fine!

**Review by JohannesStoehr — CHANGES_REQUESTED** [review]
Code looks good beside the one comment 👍

File: src/main/webapp/app/overview/course-overview.component.ts [hunk 1]
```diff
@@ -239,6 +263,10 @@ export class CourseOverviewComponent implements OnInit, OnDestroy, AfterViewInit
         return false;
     }
 
+    hasLearningGoals(): boolean {
+        return (this.course?.learningGoals?.length ?? 0) + (this.course?.prerequisites?.length ?? 0) > 0;
```
[the remark this answers is not in the document]
Comment (ge65cer) [reply src/main/webapp/app/overview/course-overview.component.ts:267]: Sadly, this does not work (type `number | undefined` is not assignable to type `boolean`).

**Review by joaquinmurcia — APPROVED** [review]
Tested on TS1. Works properly. Didn't have to refresh cache, was immediately updated.

**Review by TheTrueVester — DISMISSED** [review]
Tested on ts1. Everything worked as expected. No need to reload.

**Review by JohannesStoehr — APPROVED** [review]
Code looks good to me

File: src/main/webapp/app/overview/course-overview.component.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (Dominik-Weinzierl) [reply src/main/webapp/app/overview/course-overview.component.ts:267]: `!!this.course?.learningGoals?.length || !!this.course?.prerequisites?.length`
this should work :thinking:

[the remark this answers is not in the document]
Comment (Strohgelaender) [reply src/main/webapp/app/overview/course-overview.component.ts:267]: or `!!(this.course?.learningGoals?.length || this.course?.prerequisites?.length)`

**Review by alaraozdenler — DISMISSED** [review]
Tested on ts2, works as described

**Review by willergeorg — DISMISSED** [review]
Tested on TS5. Worked as expected. Code looks good as well. 👍

**Review by jschmautzer — DISMISSED** [review]
tested on ts1, followed the testing steps, works as described :)

File: src/main/webapp/app/overview/course-overview.component.ts [hunk 2]
```diff
@@ -201,6 +210,19 @@ export class CourseOverviewComponent implements OnInit, OnDestroy, AfterViewInit
         }
     }
 
+    loadLearningGoals() {
+        forkJoin([this.learningGoalService.getAllForCourse(this.courseId), this.learningGoalService.getAllPrerequisitesForCourse(this.courseId)]).subscribe({
```
Comment (manuelmanso) [inline src/main/webapp/app/overview/course-overview.component.ts:214]: Is there any reason why we don't get the prerequisites and the learning goals in the same call?

**Review by manuelmanso — APPROVED** [review]
Test on TS2 and works as expected! Code looks good, left just a small comment!

File: src/main/webapp/app/overview/course-overview.component.ts (hunk 2 above)
Comment (ge65cer) [reply src/main/webapp/app/overview/course-overview.component.ts:214]: That would require another endpoint on the server and a DTO class.

**Comment by krusche** [comment]
As discussed on Slack: please change the implementation so that it works without an additional caching mechanism.
The data should be loaded once when navigating into a course in case the data is not available (e.g. undefined)

**Comment by ge65cer** [comment]
@krusche That has already been changed in https://github.com/ls1intum/Artemis/pull/5241/commits/4378c2f6c8d3d764fe093cb72546a867aa9b5dea

**Comment by krusche** [comment]
Some tests fail, they need to be fixed before we can merge the PR

**Review by Dominik-Weinzierl — APPROVED** [review]
Code looks good 👍 thanks for fixing the test case ^^

**Review by babyygemperor — APPROVED** [review]
Tested on ts2, works as described

**Pull request merged**